core/src/main/scala/shapeless/lazy.scala

```diff
@@ -164,9 +164,34 @@ object Strict {
 }
 
 @macrocompat.bundle
-class LazyMacros(val c: whitebox.Context) extends CaseClassMacros {
+trait OpenImplicitMacros extends compat.CompatLite {
+  val c: whitebox.Context
+
+  import c.universe._
+
+  def openImplicitTpe: Option[Type] =
+    c.openImplicits.headOption.map(x => (x: ImplicitCandidate211).pt)
```

Comment:
Fixed that in https://github.com/milessabin/shapeless/commit/455412dd955a5a505b8727538323c17ad34360e2. I'll squash when everything's fine.